django/db/models/options.py

```diff
@@ -545,6 +545,8 @@ def get_field(self, field_name):
         """
         Return a field instance given the name of a forward or reverse field.
         """
+        if field_name == 'pk':
+            return self.pk
```

Comment:
Okay, having traced out the call graph I now believe that the only possible way that `Options.pk is None` is for the case of a model with `abstract = True` that does not otherwise have a `primary_key=True` or similar defined on it already. This is because in that case `ModelBase._prepare` will never be called which will call `Options._prepare` which will either set the key up using `self.parents` or by adding the auto `id` field which gets added via `Model.add_to_class` -> `Field.contribute_to_class` -> `Options.add_field` which will finally call `Options.setup_pk` so for the edge case of classes with `abstract = True` and no `pk` definitions `None` actually seems correct.